### PR TITLE
sysinfo.js refactor

### PR DESCRIPTION
**Pull request by codingtwinky**
fully promisifying sysinfo.js 

~~[work still under progress...]~~

ready for review

### CODE REVIEW

File: source/config.js [hunk 1]
```diff
@@ -212,3 +214,22 @@ Object.defineProperty(Error.prototype, 'toJSON', {
   },
   configurable: true
 });
+
+try {
+  module.exports.gitVersion = /.*?(\d+[.]\d+[.]\d+).*/.exec(child_process.execSync('git --version').toString())[1];
```
Comment (FredrikNoren) [inline source/config.js]: I think the reason these were async before was to shorten the startup time. Not sure if it actually is a big difference though, but might be good to look out for (i.e. if it's significantly slower maybe we should think about keeping the async version)

**Comment by FredrikNoren** [comment]
other than that one comment it looks good to me

**Comment by FredrikNoren** [comment]
Oh lol, I'm tired. Already merged :)

File: source/config.js (hunk 1 above)
Comment (codingtwinky) [inline source/config.js]: I do favor async calls above sync calls generally but for this case it was much easier to do it this way and seems like unnecessary complication for little benefits.  I don't think start up time was hindered very much.

**Comment by codingtwinky** [comment]
No please do code review even when it is merged for any and all corrections necessary.  My codes can be very sloppy sometimes... 😢

**Pull request merged**